File: app/src/main/java/com/example/my_hospital_appointments/MainPage.java
package com.example.my_hospital_appointments;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class MainPage extends AppCompatActivity {
 String userID="";
 DatabaseReference patientDetails;
       @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_page);

           Intent intent =getIntent();
           String myUsersEmail=intent.getExtras().getString("usersEmail");
          // Toast.makeText(this,"Your Email is "+myUsersEmail ,Toast.LENGTH_SHORT).show();

           patientDetails= FirebaseDatabase.getInstance().getReference("Patients");

           int Counter=myUsersEmail.length();

           for(int a=0; a<Counter; a++)
           {
               if(myUsersEmail.charAt(a)=='@')
               {
                   break;
               }
               else
               {
                  // emailFirstAppt=emailFirstAppt+myUsersEmail.charAt(a);
                   userID=userID+myUsersEmail.charAt(a);
               }
           }

           TextView myHello=(TextView)  findViewById(R.id.textViewMainPageHello );
           patientDetails.child(userID).get().addOnCompleteListener(new OnCompleteListener<DataSnapshot>() {
               @SuppressLint("SetTextI18n")
               @Override
               public void onComplete(@NonNull Task<DataSnapshot> task) {
                   if (task.isSuccessful()) {

                       DataSnapshot thisDataSnapshot = task.getResult();
                       String Name = String.valueOf(thisDataSnapshot.child("secondName").getValue());
                       myHello.setText("HELLO "+Name+",");
                   }
               }
               });

           ImageButton myContacts=(ImageButton)  findViewById(R.id.imageButtonContacts);
           myContacts.setOnClickListener(new View.OnClickListener() {
               @Override
               public void onClick(View view) {
                   Intent myIntent=new Intent(MainPage.this,Contacts.class);
                   myIntent.putExtra("usersEmail",myUsersEmail);
                   startActivity(myIntent);
               }
           });

           ImageButton bookAppointment = (ImageButton) findViewById(R.id.imageButtonBookAppointment);
        bookAppointment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent myIntent=new Intent(MainPage.this,AppointmentBooking.class);
                myIntent.putExtra("usersEmail",myUsersEmail);
                startActivity(myIntent);
            }
        });

        ImageButton consultDoctor=(ImageButton) findViewById(R.id.imageButtonPatientConsultDoctor);
        consultDoctor.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
               // Intent myIntent=new Intent(MainPage.this,PatientViewDoctorConsult.class);
               // myIntent.putExtra("usersEmail",myUsersEmail);
                Intent myIntent=new Intent(MainPage.this, Patient_Check_Doctors_to_Consult_New.class);
                myIntent.putExtra("userEmailID",userID);
                startActivity(myIntent);
            }
        });

        ImageButton medicalReports=(ImageButton) findViewById(R.id.imageButtonPatientMedicalReports);
        medicalReports.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // Intent myIntent=new Intent(MainPage.this,PatientMedicalReport.class);
                Intent myIntent=new Intent(MainPage.this,Patient_Main_Page_Activity.class);
                myIntent.putExtra("usersEmail",myUsersEmail);
                startActivity(myIntent);
            }
        });
    }
}

File: app/src/main/java/com/example/my_hospital_appointments/MainPage.kt
package com.example.my_hospital_appointments

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ImageButton
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase

class MainPage : AppCompatActivity() {
    var userID = ""
    var patientDetails: DatabaseReference? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_page)
        val intent = intent
        val myUsersEmail = intent.extras!!.getString("usersEmail")
        // Toast.makeText(this,"Your Email is "+myUsersEmail ,Toast.LENGTH_SHORT).show();
        patientDetails = FirebaseDatabase.getInstance().getReference("Patients")
        val Counter = myUsersEmail!!.length
        for (a in 0 until Counter) {
            userID = if (myUsersEmail[a] == '@') {
                break
            } else {
                // emailFirstAppt=emailFirstAppt+myUsersEmail.charAt(a);
                userID + myUsersEmail[a]
            }
        }
        val myHello = findViewById<View>(R.id.textViewMainPageHello) as TextView
        patientDetails!!.child(userID).get().addOnCompleteListener { task ->
            if (task.isSuccessful) {
                val thisDataSnapshot = task.result
                val Name = thisDataSnapshot.child("secondName").value.toString()
                myHello.text = "HELLO $Name,"
            }
        }
        val myContacts = findViewById<View>(R.id.imageButtonContacts) as ImageButton
        myContacts.setOnClickListener {
            val myIntent = Intent(this@MainPage, Contacts::class.java)
            myIntent.putExtra("usersEmail", myUsersEmail)
            startActivity(myIntent)
        }
        val bookAppointment = findViewById<View>(R.id.imageButtonBookAppointment) as ImageButton
        bookAppointment.setOnClickListener { /*
                Intent myIntent=new Intent(MainPage.this,AppointmentBooking.class);
                myIntent.putExtra("usersEmail",myUsersEmail);
                startActivity(myIntent);
                
                 */
        }
        val consultDoctor = findViewById<View>(R.id.imageButtonPatientConsultDoctor) as ImageButton
        consultDoctor.setOnClickListener { // Intent myIntent=new Intent(MainPage.this,PatientViewDoctorConsult.class);
            // myIntent.putExtra("usersEmail",myUsersEmail);
            val myIntent = Intent(this@MainPage, Patient_Check_Doctors_to_Consult_New::class.java)
            myIntent.putExtra("userEmailID", userID)
            startActivity(myIntent)
        }
        val medicalReports =
            findViewById<View>(R.id.imageButtonPatientMedicalReports) as ImageButton
        medicalReports.setOnClickListener { // Intent myIntent=new Intent(MainPage.this,PatientMedicalReport.class);
            val myIntent = Intent(this@MainPage, Patient_Main_Page_Activity::class.java)
            myIntent.putExtra("usersEmail", myUsersEmail)
            startActivity(myIntent)
        }
    }
}